src/Microsoft.DncEng.Configuration.Extensions/AppConfigurationConfigMapper.cs

```diff
@@ -0,0 +1,39 @@
+using System;
+using System.Text.RegularExpressions;
+using Azure;
+using Azure.Data.AppConfiguration;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.DncEng.Configuration.Extensions
+{
+    public static class AppConfigurationConfigMapper
+    {
+        private static readonly Regex ConfigReferenceRegex = new Regex(@"\[config\((?<key>[^])]+)\)\]");
```

Comment:
Do we... need the keys?  Isn't the key mapping the same as the "user-secrets" mapping? So if you had
```
{
  'A': {
    'B': 'some-value'
  }
```
then just having the config thing be "A.B = pizza" or whatever?

Then the setting file doesn't have to be full of "MyKey = [config(MyKey)]" redundancy.  (We have to do it for key-vaults because value keys are shared and also very limited by name)

Reply:
I wanted this to match the key vault behavior. I would have to implement another ConfigurationSource thingy if I wanted this to work the way you are suggesting. The implementation azure has for app configuration in the asp.net core IConfiguration has a horrible refreshing story.

Reply:
Ugh, that would suck...  I really like the default value being in the setting file though, otherwise it not being in the magic place will cause an unrecoverable failure.  And since the goal of this feature is to enable/disable dangerous features, having the enabling/disabling potentially causing unrecoverable errors is... unpleasant.  We should only have to use them to override the built in setting... I don't want a huge wall of other settings that we have to manage forever... with no easy way to tell who is reading them.

Reply:
Yea, I could just have it passthrough the `[config(stuff)]` value. The we can check the config value against e.g. "it must be exactly 'true'" for this thing to be on.

Reply:
If we limit them to booelans, that might work, where the returned value if "stuff" isn't defined is `false`.  I'd like to avoid yet another place we can have weird things that only fail in prod after a rollout because we forgot to define them in some third, external system